sdk/state/open.go

```diff
@@ -64,22 +70,25 @@ func (c *Channel) OpenTxs(p OpenParams) (txClose, txDecl, formation *txnbuild.Tr
 
 // ProposeOpen proposes the open of the channel, it is called by the participant
 // initiating the channel.
-func (c *Channel) ProposeOpen(p OpenParams) (Open, error) {
+func (c *Channel) ProposeOpen(p OpenParams) (OpenAgreement, error) {
 	c.startingSequence = c.initiatorEscrowAccount().SequenceNumber + 1
 
 	txClose, _, _, err := c.OpenTxs(p)
 	if err != nil {
-		return Open{}, err
+		return OpenAgreement{}, err
 	}
 	txClose, err = txClose.Sign(c.networkPassphrase, c.localSigner)
 	if err != nil {
-		return Open{}, err
+		return OpenAgreement{}, err
 	}
-	open := Open{
+	open := OpenAgreement{
 		CloseSignatures: txClose.Signatures(),
-		Asset:           p.Asset,
-		AssetLimit:      p.AssetLimit,
+		Details: OpenAgreementDetails{
```

Comment:
The golint-ci linter warning is telling you that you can do `OpenAgreementDetails(p)`.

Reply:
thx
https://github.com/stellar/experimental-payment-channels/pull/125/commits/302589e3a63055ada8fefc8fdfa44a7e088a234f